### ISSUE TITLE
[bug]: Building production package fails due to Calendar component

### ISSUE THREAD

**Issue by MilanObrenovic** [shadcn-ui/ui#5600]
### Describe the bug

Running `bun run build` (or `npm run build`) produces this error, and fails to build:

![image](https://github.com/user-attachments/assets/c90b1dde-74a6-43f6-a3ea-2876fc8b4266)

```
Type error: Object literal may only specify known properties, and 'IconLeft' does not exist in type 'Partial<CustomComponents>'.

  61 |       }}
  62 |       components={{
> 63 |         IconLeft: ({ ...props }) => <ChevronLeftIcon className="h-4 w-4" />,
     |         ^
  64 |         IconRight: ({ ...props }) => <ChevronRightIcon className="h-4 w-4" />,
  65 |       }}
  66 |       {...props}
error: script "build" exited with code 1
```

This error is fixed by reinstalling the Calendar component:

```
bunx shadcn@latest add calendar
```

But then after a while, the error appears again. No changes have been made to Calendar component, this error appears randomly. I think this needs a fix.

Additional info im using:

```
"eslint": "^8.57.0",
"eslint-config-next": "^14.2.15",
```

Newer version than these, break the calendar and other components even more, only these eslint versions are compatible together

### Affected component/components

Calendar

### How to reproduce

1. `npm run build`
2. get error from Calendar component

### Codesandbox/StackBlitz link

https://stackblitz.com/edit/stackblitz-starters-3d17cb?file=components%2Fui%2Fcalendar.tsx

### Logs

```bash
./src/components/ui/calendar.tsx:63:9
Type error: Object literal may only specify known properties, and 'IconLeft' does not exist in type 'Partial<CustomComponents>'.

  61 |       }}
  62 |       components={{
> 63 |         IconLeft: ({ ...props }) => <ChevronLeftIcon className="h-4 w-4" />,
     |         ^
  64 |         IconRight: ({ ...props }) => <ChevronRightIcon className="h-4 w-4" />,
  65 |       }}
  66 |       {...props}
error: script "build" exited with code 1
```

### System Info

```bash
iMac, 3.8GHz, 8-Core Intel Core i7, AMD Radeon Pro 5500 XT 8GB, 40GB RAM 2133MHz DDR4, Sonoma 14.6.1

Brave Browser v1.70.117
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by kasperadk** [comment]
Getting the same error.

**Comment by MilanObrenovic** [comment]
> Getting the same error.

Seems like this issue occurs due to the new version of `react-day-picker`, it has some breaking changes to this component. See this issue https://github.com/shadcn-ui/ui/issues/5417

This error is solved when i set

```
"react-day-picker": "^8.10.1"
```

Still, shadcn needs an update for the Calendar component to be compatible with the new version of react-day-picker

**Comment by bdlowery** [comment]
Look at the docs. This is what you need to change -

https://daypicker.dev/upgrading#7-update-your-custom-components

**Comment by emanuelsan** [comment]
@bdlowery points in the right direction for temporarily fixing this issue.

However, directly applying the recommendation from `react-day-picker` is not sufficient. This is because the components mentioned in [the fix](https://daypicker.dev/upgrading#7-update-your-custom-components) - `<MyLeftIcon />` and `<Chevron />` - are not present in the shadcn `calendar` component.

Fortunately, the fix is very straightforward:

```jsx
<DayPicker
    // Other attributes here...
    components={{
        Chevron: (props) => {
              if (props.orientation === "left") {
                  return <ChevronLeftIcon {...props} />;
              }
              return <ChevronRightIcon {...props} />;
        },
    }}
/>
```

Basically, just use the icons that are already imported in the calendar component: `<ChevronLeftIcon />` and `<ChevronRightIcon />`. This should fix the type errors.

Still, as @MilanObrenovic said, the calendar component needs a proper fix.

**Comment by gpbl** [comment]
👋 `react-day-picker` maintainer here! What should I do to help shadcn-ui users with the calendar? I understand the upgrade to DayPicker v9 is not working as expected.

1. Reintroduce deprecated props to DayPicker to maintain backward compatibility with `shadcn-ui`. (will the style still work?) _or_
2. Update shadecn-ui to use the latest DayPicker, however it seems  https://github.com/shadcn-ui/ui/issues/5417 is not receiving much attention.

> Fortunately, the fix is very straightforward:

Thanks @emanuelsan, I've updated the DayPicker upgrading docs.

**Comment by edtorba** [comment]
> 👋 `react-day-picker` maintainer here! What should I do to help shadcn-ui users with the calendar? I understand the upgrade to DayPicker v9 is not working as expected.
> 
> 1. Reintroduce deprecated props to DayPicker to maintain backward compatibility with `shadcn-ui`. (will the style still work?) _or_
> 2. Update shadecn-ui to use the latest DayPicker, however it seems  [[bug]: breaking changes for calendar using react-day-picker 9.1.4 #5417](https://github.com/shadcn-ui/ui/issues/5417) is not receiving much attention.
> 
> > Fortunately, the fix is very straightforward:
> 
> Thanks @emanuelsan, [I've updated](https://github.com/gpbl/react-day-picker/commit/c9d0e4186b1f438e2756cc14e9445737ea322c59) the DayPicker upgrading docs.

I'd vote for the second option. `react-day-picker` had a major version release, so breaking changes can be expected. The best solution is to sort the `calendar.tsx` file.

**Comment by comphonia** [comment]
Update Jan 2025. This is still an issue, If you're looking for a drop-in replacement, consider this project: https://date-picker.luca-felix.com/

Just follow the "Update the Calendar component" step and replace the cod. 

It just works!!! and you get a nice little feature (The ability to select the year 😊)

**Comment by pimentellima** [comment]
> Update Jan 2025. This is still an issue, If you're looking for a drop-in replacement, consider this project: https://date-picker.luca-felix.com/
> 
> Just follow the "Update the Calendar component" step and replace the cod.
> 
> It just works!!! and you get a nice little feature (The ability to select the year 😊)

this is better than the original

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to one year of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you. (This is an automated message)

**Comment by shadcn** [comment]
This issue has been automatically closed due to one year of inactivity. If you’re still experiencing a similar problem or have additional details to share, please open a new issue following our current issue template. Your updated report helps us investigate and address concerns more efficiently. Thank you for your understanding! (This is an automated message)
